pulsar-broker/src/test/java/org/apache/pulsar/client/impl/PatternTopicsConsumerImplTest.java

```diff
@@ -679,6 +681,58 @@ public void testAutoSubscribePatterConsumerFromBrokerWatcher(boolean delayWatchi
         }
     }
 
+    @DataProvider(name= "partitioned")
+    public Object[][] partitioned(){
+        return new Object[][]{
+                {true},
+                {false}
+        };
+    }
+
+    @Test(timeOut = testTimeout, dataProvider = "partitioned")
+    public void testPreciseRegexpSubscribe(boolean partitioned) throws Exception {
```

Comment:
Also need to test case when  `enableBrokerSideSubscriptionPatternEvaluation=false`.

Reply:
After enabling `enableBrokerSideSubscriptionPatternEvaluation `, the feature `TopicListWatcher` will be disabled, too. I will create a separate discussion to cover this case.

Reply:
Added the test: `testPreciseRegexpSubscribeDisabledTopicWatcher`